google/cloud/spanner_dbapi/cursor.py

```diff
@@ -290,7 +290,7 @@ def fetchall(self):
         res = []
         try:
             for row in self:
-                if not self.connection.autocommit:
+                if not self.connection.autocommit and not self.connection.read_only:
                     self._checksum.consume_result(row)
                 res.append(row)
         except Aborted:
```

Comment:
Add an `if` statement to this and other `except Aborted` blocks to prevent `retry_transaction` from being called when it is a read-only transaction.

Reply:
This comment has been marked as resolved, but no changes were made. Do you disagree with my suggestion, or is this something you are still working on?